### PR TITLE
MRG: Fix drop log plot

### LINKED ISSUE #2730: epochs drop log plot % unclear
a student made me realize that the percentages we display in plot drop log is not very understandable. I could not explain it myself :)

here is plot 

figure_1

obtained with the script below. Why do I see 21% rejected but only channels responsible for less than 10% of the bad epochs?

@Eric89GXL can you make it easier to understand? How about adding to the title the number of rejected epochs (not just the %)?

```
import mne
from mne.datasets import sample

data_path = sample.data_path()
raw_fname = data_path + '/MEG/sample/sample_audvis_raw.fif'
event_fname = data_path + '/MEG/sample/sample_audvis_raw-eve.fif'
tmin, tmax = -0.2, 0.5

# Select events to extract epochs from.
event_id = {'Auditory/Left': 1, 'Auditory/Right': 2}

#   Setup for reading the raw data
raw = mne.io.Raw(raw_fname)
events = mne.read_events(event_fname)

#   Set up pick list: EEG + STI 014 - bad channels (modify to your needs)
include = []  # or stim channels ['STI 014']
raw.info['bads'] = ['MEG 2443', 'EEG 053']  # set bads

# pick EEG and MEG channels
picks = mne.pick_types(raw.info, meg=True, eeg=True, stim=False, eog=True,
                       include=include, exclude='bads')
# Read epochs
epochs = mne.Epochs(raw, events, event_id, tmin, tmax, picks=picks,
                    baseline=(None, 0),
                    reject=dict(grad=4000e-13, mag=4e-12, eog=150e-6),
                    preload=True)

epochs.drop_bad_epochs()
epochs.plot_drop_log(subject='sample')
```

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2730.

@agramfort works for me, ready for review/merge. Didn't add a test because it's for a `viz` function.

### CODE REVIEW

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -223,7 +198,11 @@ def plot_drop_log(drop_log, threshold=0, n_max_plot=20, subject='Unknown',
     if perc < threshold or len(ch_names) == 0:
         plt.text(0, 0, 'No drops')
         return fig
-    counts = 100 * np.array(list(scores.values()), dtype=float) / len(drop_log)
+    n_used = 0
+    for d in drop_log:
+        if len(d) == 0 or any(ch not in ignore for ch in d):
+            n_used += 1
+    counts = 100 * np.array(list(scores.values()), dtype=float) / n_used
```
Comment (jasmainak) [inline mne/viz/epochs.py]: this would be a Bug fix, no? I would add it somewhere in the whats_new.rst

Comment (Eric89GXL) [inline mne/viz/epochs.py]: ahh yeah forgot that will do

File: mne/viz/epochs.py [hunk 2]
```diff
@@ -223,7 +198,11 @@ def plot_drop_log(drop_log, threshold=0, n_max_plot=20, subject='Unknown',
     if perc < threshold or len(ch_names) == 0:
         plt.text(0, 0, 'No drops')
         return fig
-    counts = 100 * np.array(list(scores.values()), dtype=float) / len(drop_log)
+    n_used = 0
+    for d in drop_log:
```
Comment (jasmainak) [inline mne/viz/epochs.py]: can you add a comment which indicates that `d` is the drop log for a particular epoch?

Comment (jasmainak) [inline mne/viz/epochs.py]: I had to do some thinking before realizing what `n_used` was doing ...

**Comment by jona-sassenhagen** [comment]
Don't you want a bit of space between the plot's upper axis and the largest bar?

**Comment by jasmainak** [comment]
other than my minor comments, LGTM. +1 for merge once Travis is happy

**Comment by Eric89GXL** [comment]
@jona-sassenhagen perhaps, but I'm happy to take whatever `matplotib` wants to do with the y-axis limits. Trying to tweak it is not worth the effort to me, but feel free to ENH later if you get bored or sufficiently annoyed :)

**Comment by jaeilepp** [comment]
@Eric89GXL . You can do something like ``plt.gca().set_ylim((0, max(counts) * 1.1))`` to make some space. I think I've done something like this before.

**Comment by Eric89GXL** [comment]
@jaeilepp feel free to open a PR, it's an orthogonal change to this one

**Comment by Eric89GXL** [comment]
(I don't want to hold up this bugfix by tweaking and settling on aesthetic changes)

**Comment by jona-sassenhagen** [comment]
Doesn't bother me really as I never use this kind of function.

**Comment by Eric89GXL** [comment]
Travis is happy

File: doc/whats_new.rst [hunk 3]
```diff
@@ -53,6 +53,8 @@ BUG
 
     - Fix channel name pick in :func:`mne.Evoked.get_peak` method by `Alex Gramfort`_
 
+    - Fix drop percentages to take into account ``ignore`` option in :func:`mne.viz.plot_drop_log` and :func:`mne.Epochs.plot_drop_log` by `Eric Larson`_.
```
Comment (jasmainak) [inline doc/whats_new.rst]: slightly orthogonal question but why are both functions public? Shouldn't we have only the Epochs function in the public API?

File: mne/viz/epochs.py [hunk 4]
```diff
@@ -223,7 +198,11 @@ def plot_drop_log(drop_log, threshold=0, n_max_plot=20, subject='Unknown',
     if perc < threshold or len(ch_names) == 0:
         plt.text(0, 0, 'No drops')
         return fig
-    counts = 100 * np.array(list(scores.values()), dtype=float) / len(drop_log)
+    n_used = 0
+    for d in drop_log:  # "d" is the list of drop reasons for each epoch
```
Comment (jasmainak) [inline mne/viz/epochs.py]: excellent!

File: doc/whats_new.rst (hunk 3 above)
Comment (Eric89GXL) [inline doc/whats_new.rst]: The former is a function and can operate on list of lists, second is a method of the `Epochs` class, not a function in the `mne.epochs` namespace. Make sense?

**Comment by jasmainak** [comment]
I'm good to merge. One more +1? @jaeilepp or @agramfort ?

**Comment by jaeilepp** [comment]
LGTM

**Comment by jasmainak** [comment]
Thanks @Eric89GXL for the quick fix !

**Comment by Eric89GXL** [comment]
Thanks for the quick reviews

**Comment by agramfort** [comment]
thx @Eric89GXL !

**Pull request merged**